src/engine/QueryExecutionContext.h

```diff
@@ -112,16 +119,22 @@ class QueryExecutionContext {
     return _sortPerformanceEstimator;
   }
 
-  void readCostFactorsFromTSVFile(const string& fileName) {
-    _costFactors.readFromFile(fileName);
-  }
-
   [[nodiscard]] double getCostFactor(const string& key) const {
     return _costFactors.getCostFactor(key);
   };
 
   ad_utility::AllocatorWithLimit<Id> getAllocator() { return _allocator; }
 
+  void signalQueryUpdate(const RuntimeInformation& runtimeInformation) {
+    webSocketManager_.addQueryStatusUpdate(
+        owningQueryId_.toQueryId(),
+        nlohmann::ordered_json(runtimeInformation).dump());
+  }
+
+  ~QueryExecutionContext() {
+    webSocketManager_.releaseQuery(owningQueryId_.toQueryId());
+  }
```

Comment:
Hmm currently the `OwningQueryId` class is completely independent of a websocket manager. Not sure if it would be the right move to change that. So conceptually I thought the `QueryExecutionContext` would be the right place to perform this kind of cleanup because here all the components come together.